src/vs/workbench/parts/terminal/electron-browser/terminalActions.ts

```diff
@@ -653,7 +654,19 @@ export class RunActiveFileInTerminalAction extends Action {
 			this.notificationService.warn(nls.localize('workbench.action.terminal.runActiveFile.noFile', 'Only files on disk can be run in the terminal'));
 			return TPromise.as(void 0);
 		}
-		instance.sendText(uri.fsPath, true);
+		let uriPath: string = uri.fsPath;
+		const hasSpace = uriPath.indexOf(' ') !== -1;
+		if (hasSpace && isWindows) {
+			uriPath = '"' + uriPath + '"';
+		} else if (!isWindows) {
+			const hasDoubleQuote = uriPath.indexOf('"') !== -1;
+			if (!hasSpace && hasDoubleQuote) {
+				uriPath = '\'' + uriPath + '\'';
+			} else if (hasSpace) {
+				uriPath = uriPath.replace(' ', '\\ ');
```

Comment:
`.replace` only replaces the first instance.

I think we also need to handle `\\` -> `\\\\`